gobblin-modules/gobblin-sql/src/main/java/org/apache/gobblin/source/jdbc/JdbcExtractor.java

```diff
@@ -959,6 +959,10 @@ public long getCount(CommandOutput<?, ?> response) throws RecordCountException {
       int batchSize = this.workUnitState.getPropAsInt(ConfigurationKeys.SOURCE_QUERYBASED_FETCH_SIZE, 0);
       batchSize = (batchSize == 0 ? ConfigurationKeys.DEFAULT_SOURCE_FETCH_SIZE : batchSize);
 
+      String sourceConnProps = this.workUnitState.getProp(ConfigurationKeys.SOURCE_CONN_PROPERTIES);
+      boolean convertZeroDateTimeToNull = sourceConnProps != null
```

Comment:
I think here we can generalize this a bit more rather than keep it only to the zeroDateTime behavior, to support the other zeroDateTime properties if there are any users. We can check that sourceConnProps contains zeroDateTimeBehavior, then parse it as timestamp and return the result. This way it can handle the `ROUND` case